Backend/Remora.Discord.Gateway/DiscordGatewayClient.cs

```diff
@@ -300,13 +344,11 @@ public async Task<Result> RunAsync(CancellationToken stopRequested)
         }
         finally
         {
-            _sessionID = null;
```

Comment:
Should this not null out the session ID in the session information or replace the session information entirely?

Reply:
Elaborate?

Reply:
Well, the whole point of this section is to reset the internal state to a clean slate when session resumption is explicitly not possible or desired. As the code is, the old session information will always hang around.

Reply:
Right. In theory the only time that this should happen is when Discord sends `OP9` (Reconnect) with `d = false` (which indicates that the session cannot be resumed).